haystack/pipeline.py

```diff
@@ -239,6 +240,34 @@ def run(self, query: str, filters: Optional[Dict] = None, top_k_retriever: int =
         return output
 
 
+class SearchSummarizationPipeline(BaseStandardPipeline):
+    def __init__(self, summarizer: BaseSummarizer, retriever: BaseRetriever):
+        """
+        Initialize a Pipeline that retrieves documents for a query and then summarizes those documents.
+
+        :param summarizer: Summarizer instance
+        :param retriever: Retriever instance
+        """
+        self.pipeline = Pipeline()
+        self.pipeline.add_node(component=retriever, name="Retriever", inputs=["Query"])
+        self.pipeline.add_node(component=summarizer, name="Summarizer", inputs=["Retriever"])
+
+    def run(
+        self,
+        query: str,
+        filters: Optional[Dict] = None,
+        top_k_retriever: int = 10,
+        generate_single_summary: bool = False
+    ):
+        output = self.pipeline.run(
+            query=query, filters=filters, top_k_retriever=top_k_retriever, generate_single_summary=generate_single_summary
+        )
+
+        # TODO do conversion of docs to answers here?
```

Comment:
We could return `Document` objects from the summarizer and format them to "answers" here ...

Reply:
Do you want change in this PR or in separate PR?

Reply:
I would do it directly in this PR to avoid any breaking changes ...